examples/gno.land/p/demo/timelock/timelock.gno

```diff
@@ -0,0 +1,173 @@
+// Package timelock provides a library for scheduling, cancelling, and
+// executing time-locked operations in Gno. It ensures that
+// operations are only carried out after a specified delay and offers
+// mechanisms for managing and verifying the status of these operations.
+// This package leverages an AVL tree for efficient management of timestamps
+// and integrates role-based access control for administrative tasks.
+//
+// Example Usage:
+//
+//  import "gno.land/p/demo/timelock"
+//  import "gno.land/p/demo/accesscontrol"
+//
+//  Initialize timelock utility with an AVL tree and access control.
+//  timestamps := avl.NewTree()
+//  adminRole := accesscontrol.NewRoleData("admin", std.Address("admin-address"))
+//  timelockUtil := timelock.NewTimelockUtil(timestamps, adminRole)
+//
+//  Schedule an operation with a delay of 60 seconds and a minimum delay of 30 seconds.
+//  id := seqid.ID()
+//  timelockUtil.Schedule(id, 60, 30)
+//
+//  Check if an operation is pending.
+//  isPending := timelockUtil.IsOperationPending(id)
+//
+//  Execute the operation when it is ready.
+//  if timelockUtil.IsOperationReady(id) {
+//      timelockUtil.Execute(id)
+//  }
+//
+//  Update the minimum delay for future operations.
+//  var minDelay int64 = 30
+//  timelockUtil.UpdateDelay(&minDelay, 45)
+
+package timelock // import "gno.land/p/demo/timelock"
+
+import (
+	"std"
+	"strconv"
+	"time"
+	"ufmt"
+
+	"gno.land/p/demo/accesscontrol"
+	"gno.land/p/demo/avl"
+	"gno.land/p/demo/seqid"
+)
+
+// Marks operations completed
+const DoneTimestamp = 1
+
+// Represents the status of a planned operation
+type OperationState int
+
+const (
+	Unset OperationState = iota
+	Waiting
+	Ready
+	Done
+)
+
+// TimelockUtil stores the necessary parameters for the timelock operations
+type TimelockUtil struct {
```

Comment:
Thoughts on naming this `TimeLockUtil`? The current naming seems a bit strange as though "Timelock" were one word rather than two words.

Reply:
You're right, that's better. 

- https://github.com/gnolang/gno/pull/2307/commits/863d1c8730f0109f88e4177ffd9fd2ec96deab21